components/menu/MenuItem.tsx

```diff
@@ -66,7 +66,7 @@ export default class MenuItem extends React.Component<MenuItemProps> {
         {...rest}
         className={classNames(
           {
-            [`${prefixCls}-item-danger`]: danger,
+            [`${prefixCls}-item-danger`]: this.props.disabled ? false : danger,
             [`${prefixCls}-item-only-child`]: (icon ? childrenLength + 1 : childrenLength) === 1,
```

Comment:
I think we can solve the problem here by increasing the css selector weights.

https://github.com/ant-design/ant-design/blob/f4d011623ebcb0eafe5e3ac9228d3f7b681c5152/components/dropdown/style/index.less#L224-L237

what do you think?

```diff
-      &-disabled {
+      &&-disabled {
```

Reply:
Thank you 

I tried this out in version 5x, but it didn't work out.
It will break the disable option as shown in below (Image-1). But yes this is much cleaner and better i guess but i couldn't make it i tried a few more ways of increasing the **CSS specificity**, but they didn't work.

![Image-1](https://user-images.githubusercontent.com/75088294/206263342-74265eb0-3766-4d8a-8ee0-cdb651c1b5a6.png)

![code-1](https://user-images.githubusercontent.com/75088294/206263468-5e6a98e3-4835-4d49-aa27-18e331e6209a.png)

Reply:
What about adding `!important` statement to `color` and `background-color` ?

Reply:
The use of `!important ` is bad, in 5.x it can be changed to this

```diff
 
-            '&-disabled': {
+            [`&${menuCls}-item-disabled`]: {
               color: colorTextDisabled,

```

and switch the order of returns

```diff
     return [
+      genStatusStyle(dropdownToken),
       genBaseStyle(dropdownToken),
       genButtonStyle(dropdownToken),
-      genStatusStyle(dropdownToken),
     ];
```

Reply:
this is just my suggestion, still need official help to review @afc163

Reply:
@Wxh16144's suggestion is what I think too.